### PR TITLE
Update "Bootstrapping" chapter

### PR DESCRIPTION
**Pull request by linawolf**
Replaces #897

### CODE REVIEW

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 1]
```diff
@@ -7,60 +7,79 @@ Bootstrapping
 =============
 
 TYPO3 CMS has a clean bootstrapping process driven mostly
-by class :php:`\TYPO3\CMS\Core\Core\Bootstrap`. This class
-contains a host of methods each responsible for a little
-step along the initialization of a full TYPO3 process,
-be it the backend or other contexts.
+by class :php:`\TYPO3\CMS\Core\Core\Bootstrap`. This class is initialized by
+calling  :php:`Bootstrap::init()` and serves as an entry point for later calling
+an Application class, depending on the TYPO3_MODE and several context-dependant
```
Comment (mbrodala) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:12]: I think we should keep regular terms like `application` in lowercase unless we refer to a specific "thing" which will then very likely be a class name.

Also we should not refer to `TYPO3_MODE` anymore which is deprecated in TYPO3v11.

**Review by mbrodala — CHANGES_REQUESTED** [review]

**Comment by linawolf** [comment]
What is the status of this @mbrodala, can it be merged?

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 2]
```diff
@@ -7,60 +7,78 @@ Bootstrapping
...
-contains a host of methods each responsible for a little
-step along the initialization of a full TYPO3 process,
-be it the backend or other contexts.
+by class :php:`\TYPO3\CMS\Core\Core\Bootstrap`. This class is initialized by
+calling  :php:`Bootstrap::init()` and serves as an entry point for later calling
+an application class, depending on several context-dependant constraints.
 
-Some contexts add their own bootstrap class (like the command
-line, which additionally requires :php:`\TYPO3\CMS\Core\Core\CliBootstrap`).
+Each application class registers Request Handlers to the TYPO3 Bootstrap to
```
Comment (mbrodala) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:14]:
```suggestion
Each application class registers request handlers to
```

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 3]
```diff
@@ -7,60 +7,78 @@ Bootstrapping
...
-   This bootstrapping API is internal and may change at any time in the near future
-   even in minor updates. It is thus discouraged to use it in third party code.
-   Use this class only if other extensibility possibilities such as
-   :ref:`Hooks <hooks>`, Signals or :ref:`XCLASS <xclasses>`
-   are not enough to reach your goals.
+\\TYPO3\\CMS\\Frontend\\Http\\Application
+-----------------------------------------
 
+This class handles all incoming web requests coming through :file:`index.php`
+in the main directory. It handles all TSFE and eID requests.
```
Comment (mbrodala) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:28]:
```suggestion
in the public web directory. It handles all regular page (TSFE) and eID requests.
```

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 4]
```diff
@@ -7,60 +7,78 @@ Bootstrapping
...
-      ->setRequestType(TYPO3_REQUESTTYPE_BE | (!empty($_GET['ajaxID']) ? TYPO3_REQUESTTYPE_AJAX : 0))
-      ->baseSetup($this->entryPointLevel);
+This class handles all incoming web requests for any regular backend call
+inside :file:`typo3/\*`.
 
-   // Redirect to Install Tool if base configuration is not found
-   if (!$this->bootstrap->checkIfEssentialConfigurationExists()) {
-      $this->bootstrap->redirectToInstallTool($this->entryPointLevel);
-   }
+Its :php:`TYPO3\CMS\Backend\Http\RequestHandler` is used for all Backend
```
Comment (mbrodala) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:40]:
```suggestion
Its :php:`TYPO3\CMS\Backend\Http\RequestHandler` is used for all backend
```

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 5]
```diff
@@ -7,60 +7,78 @@ Bootstrapping
...
+requests, including AJAX routes. If a get/post parameter "route" is set, the
+Backend Routing is called by the :php:`RequestHandler` and
```
Comment (mbrodala) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:42]:
```suggestion
backend routing is called by the :php:`RequestHandler` and
```

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 6]
```diff
@@ -7,60 +7,78 @@ Bootstrapping
...
+searches for a matching route inside the Router. The corresponding controller
```
Comment (mbrodala) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:43]:
```suggestion
searches for a matching route inside the router. The corresponding controller
```

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 7]
```diff
@@ -7,60 +7,78 @@ Bootstrapping
...
+redirects to the TYPO3 Install Tool.
 
-   $this->bootstrap->configure();
 
-   ###
+\\TYPO3\\CMS\\Core\\Console\\CommandApplication
+-----------------------------------------------
+
+This class is the entry point for the TYPO3 command line for console commands.
+In addition to a simple Symfony command, this also sets up a CLI user.
```
Comment (mbrodala) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:54]: What is the "simple Symfony command"?

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 8]
```diff
@@ -7,60 +7,78 @@ Bootstrapping
...
+
+
+\\TYPO3\\CMS\\Install\\Http\\Application
+----------------------------------------
+
+The install tool :php:`Application` only runs with a very limited bootstrap
+set up. The failsafe package manager does not take
+the :php:`ext_localconf.php` scripts of installed extensions into account.
```
Comment (mbrodala) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:62]:
```suggestion
the :php:`ext_localconf.php` of installed extensions into account.
```

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 9]
```diff
@@ -7,60 +7,78 @@ Bootstrapping
...
+
+.. warning::
+
+   This bootstrapping API is internal and may change at any time in the near future
+   even in minor updates. It is thus discouraged to use it in third party code.
+   Use this class only if other extensibility possibilities such as
+   :ref:`Hooks <hooks>`, Signals or :ref:`XCLASS <xclasses>`
```
Comment (mbrodala) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:69]:
```suggestion
   :ref:`Events <eventdispatcher>`, :ref:`Hooks <hooks>`, or :ref:`XCLASS <xclasses>`
```

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 10]
```diff
@@ -7,60 +7,78 @@ Bootstrapping
...
-Note that most methods of the Bootstrap class must be called in a precise order.
-It is perfectly possible to define one's own bootstrapping process, but care
-should be taken about the call order.
+.. code-block:: php
 
-Also note that all bootstrapping methods return the instance of the
-Bootstrap class itself, allowing calls to be chained.
+   // Set up the application for the backend
+   call_user_func(function () {
+       $classLoader = require dirname(__DIR__).'/vendor/autoload.php';
```
Comment (mbrodala) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:78]:
```suggestion
       $classLoader = require dirname(__DIR__) . '/vendor/autoload.php';
```

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 11]
```diff
@@ -72,6 +90,10 @@ Whenever a call to TYPO3 CMS is made, the application goes through a
 bootstrapping process managed by a dedicated API. This process is also
 used in the frontend, but only the backend process is described here.
 
+
+Classes involved in the backend bootstrapping process are :php:`\TYPO3\CMS\Core\Core\Bootstrap`,
```
Comment (mbrodala) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:94]: There are *many* more classes involved so what was the intention of mentioning these?

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 12]
```diff
@@ -80,43 +102,36 @@ Classes involved in the backend bootstrapping process are :php:`\TYPO3\CMS\Core\
...
+   ajax or Install Tool request.
 
 -  :php:`SystemEnvironmentBuilder::defineBaseConstants()` defines
    constants containing values such as the current version number,
    blank character codes and error codes related to services.
 
--  :php:`SystemEnvironmentBuilder::definePaths()` defines constants
-   containing paths to various parts of the TYPO3 installation like
+-  :php:`SystemEnvironmentBuilder::initializeEnvironment()` initializes the
+   Environment class that points to various parts of the TYPO3 installation like
```
Comment (mbrodala) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:127]:
```suggestion
   :php:`Environment` class that points to various parts of the TYPO3 installation like
```

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 13]
```diff
@@ -80,43 +102,36 @@ Classes involved in the backend bootstrapping process are :php:`\TYPO3\CMS\Core\
...
    the absolute path to the :file:`typo3` directory or the absolute
    path to the installation root.
 
--  :php:`SystemEnvironmentBuilder::checkMainPathsExist()` checks if
-   expected paths like :file:`typo3` or :file:`index.php` exist. If that
-   is not the case, the process will quit immediately.
+-  :php:`SystemEnvironmentBuilder::calculateScriptPath()` calculates the script
+   path. This is the absolute path to the entry script. This can be something
+   like '.../public/index.php' or '.../public/typo3/index.php' for web calls,
+   or '.../bin/typo3' or similar for cli calls.
```
Comment (mbrodala) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:134]:
```suggestion
   or '.../bin/typo3' or similar for CLI calls.
```

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 14]
```diff
@@ -125,69 +140,63 @@ worth taking a look into the following methods:
...
----------------------
+-  :php:`Bootstrap::createConfigurationManager()` creates the Configuration
+   Manager which is then populated with the the main configuration ("TYPO3_CONF_VARS").
 
-Next :php:`Bootstrap::configure()` is called which in turn triggers
-a whole new series of configuration. This is actually a major step,
-with too many actions to detail efficiently here. However here is the
-list of the most important stuff happening at this point:
+-  :php:`$builder->createDependencyInjectionContainer()` creates a dependency
+   injection container which is later returned by :php:`Bootstrap::init()`.
```
Comment (mbrodala) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:162]:
```suggestion
-  :php:`$builder->createDependencyInjectionContainer()` creates a :ref:`dependency
   injection <dependency-injection>` container which is later returned by :php:`Bootstrap::init()`.
```

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 15]
```diff
@@ -125,69 +140,63 @@ worth taking a look into the following methods:
...
 
--  the main configuration ("TYPO3_CONF_VARS") is loaded
+-  The Caching Framework and the Package Management are set up.
```
Comment (mbrodala) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:164]:
```suggestion
-  The caching framework and the package management are set up.
```

**Review by mbrodala — CHANGES_REQUESTED** [review]
Almost there. :-)

In general I'd recommend to always use lowercase for all words except for proper names and unique things.

File: Documentation/ApiOverview/Bootstrapping/Index.rst (hunk 7 above)
Comment (linawolf) [reply Documentation/ApiOverview/Bootstrapping/Index.rst:54]: No idea, do you know? Or should we just remove the sentence? (Please note I only try to get this old PR merged, I didn't write the texts in the first place)

File: Documentation/ApiOverview/Bootstrapping/Index.rst (hunk 11 above)
Comment (linawolf) [reply Documentation/ApiOverview/Bootstrapping/Index.rst]: Removed the sentence

**Comment by linawolf** [comment]
@mbrodala ok, included your changes. Can we merge now?

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 16]
```diff
@@ -7,60 +7,78 @@ Bootstrapping
...
+redirects to the TYPO3 Install Tool.
 
-   $this->bootstrap->configure();
 
-   ###
+\\TYPO3\\CMS\\Core\\Console\\CommandApplication
+-----------------------------------------------
 
+This class is the entry point for the TYPO3 command line for console commands.
+In addition to a simple Symfony command, this also sets up a CLI user.
```
Comment (mbrodala) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:54]:
```suggestion
In addition to registering all available commands, this also sets up a CLI user.
```

**Review by mbrodala — CHANGES_REQUESTED** [review]

**Review by mbrodala — APPROVED** [review]

**Pull request merged**